bin/desi_qso_mgii_afterburner

```diff
@@ -0,0 +1,250 @@
+
+
+def save_dataframe_to_fits(dataframe, filename, DESI_TARGET, clobber=True):
+    """
+    Save info from pandas dataframe in a fits file. Need to write the dtype array
+    because of the list in the pandas dataframe (no other solution found)
+    Args:
+        dataframe (pandas dataframe): dataframe containg the all the necessary QSO info
+        filename (str):  name of the fits file
+        DESI_TARGET (str): name of DESI_TARGET for the wanted version of the target selection
+        clobber (bool): overwrite the fits file defined by filename ?
+    Returns:
+        None
+    """
+    # Ok we cannot use dataframe.to_records() car les coeffs/c_lines sont sauvegarder sous forme de list de type objet et ne peux pas etre convertit ..
+    data = np.zeros(dataframe.shape[0], dtype=[('TARGETID', 'i8'), ('RA','f8'), ('DEC','f8'), ('Z_RR','f8'), ('ZERR', 'f4'), ('IS_QSO_MGII', '?'),
+                                               (DESI_TARGET, 'i8'), ('SPECTYPE', 'U10'), #('COEFFS', ('f4', 10)),
+                                               ('DELTA_CHI2', 'f4'), ('A', 'f4'), ('SIGMA', 'f4'), ('B', 'f4'),
+                                               ('VAR_A', 'f4'), ('VAR_SIGMA', 'f4'), ('VAR_B', 'f4')])
+
+    data['TARGETID'] = dataframe['TARGETID']
+    data['RA'] = dataframe['RA']
+    data['DEC'] = dataframe['DEC']
+    data['Z_RR'] = dataframe['Z_RR']
+    data['ZERR'] = dataframe['ZERR']
+    data['IS_QSO_MGII'] = dataframe['IS_QSO_MGII']
+    data[DESI_TARGET] = dataframe[DESI_TARGET]
+    data['SPECTYPE'] = dataframe['SPECTYPE']
+    #data['COEFFS'] = np.array([np.array(dataframe['COEFFS'][i]) for i in range(dataframe.shape[0])])
+
+    data['DELTA_CHI2'] = dataframe['DELTA_CHI2']
+    data['A'] = dataframe['A']
+    data['SIGMA'] = dataframe['SIGMA']
+    data['B'] = dataframe['B']
+    data['VAR_A'] = dataframe['VAR_A']
+    data['VAR_SIGMA'] = dataframe['VAR_SIGMA']
+    data['VAR_B'] = dataframe['VAR_B']
+
+    fits = fitsio.FITS(filename, 'rw', clobber=clobber)
+    fits.write(data, extname='MGII')
+    if clobber:
+        log.info(f'OVERWRITE the file : {filename}')
+    else:
+        log.info(f'EXPAND the file : {filename}')
+    fits.close()
+
+    return
+
+
+if __name__ == "__main__":
+
+    args = collect_argparser()
+
+    # Param for the MgII fitter see desispec/py/desispec/mgii_afterburner.py for additional informations
+    param_mgii_fitter = {'lambda_width':args.lambda_width, 'max_sigma':args.max_sigma, 'min_sigma':args.min_sigma,
+                         'min_deltachi2':args.min_deltachi2, 'min_signifiance_A':args.min_signifiance_A, 'min_A':args.min_A}
+
+
+    if os.path.isfile(args.coadd) & os.path.isfile(args.zbest):
+        # open best fit file generated by redrock
+        zbest_file = fitsio.FITS(args.zbest)
+        zbest = zbest_file[1]
+        fibermap = zbest_file[2]
+        index_sorted = np.argsort(fibermap['TARGETID'][:500]) # in zbest targetid are sorted, it is not the case in the fibermap
```

Comment:
In the fibermap I think this works by luck of how they are written, but not for a general order of the rows.  This would be more robust and could handle any ordering of repeated TARGETID, and well as any number of TARGETIDs (e.g. the healpix grouping will not have multiple of 500s and different TARGETIDs could appear a different number of times):
```
index_sorted = np.unique(fibermap['TARGETID'], return_index=True)[1]
```

Reply:
Ok, I supposed that the form will be always the same (ie) by petal and then with only 500 objects in each redrock file which are sorted by redrock by TARGETID in the zbest file. So I will change !